volmigrate/migration.go

```diff
@@ -0,0 +1,41 @@
+package volmigrate
+
+import (
+	"github.com/contiv/errored"
+	"github.com/contiv/volplugin/volmigrate/backend"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+// Migrations is used to sort by the version number.
+type Migrations []*Migration
+
+func (m Migrations) Len() int           { return len(m) }
+func (m Migrations) Swap(i, j int)      { m[i], m[j] = m[j], m[i] }
+func (m Migrations) Less(i, j int) bool { return m[i].Version < m[j].Version }
+
+// Migration represents a migration that volmigrate can run.
+type Migration struct {
+	Version     int64  // each new migration increments this value, must be >= 1
+	Description string // e.g., "volplugin 1.3 release"
+	runner      func(backend.Backend) error
+}
+
+// Run executes the migration using a supplied backend.Backend
+func (m *Migration) Run(backend backend.Backend) error {
+	log.WithFields(log.Fields{
```

Comment:
ah ok. Let's leave this out then; I want to do more structured logging but would like to also keep things consistent. Once we get the chance we will.